File: tests/pp.m
clear all; close all; clc; 


filename = "compEigenVsfMatrix";

x = dlmread(filename);
x = x(2:end, :);
disp(x);
n = x(:,1);


figure(1)
hold on; 

loglog((n), (x(:, 2)), '*-', 'DisplayName', 'fullPivLU');

loglog((n), (x(:, 2)), '*-', 'DisplayName', 'fullPivLU');
loglog((n), (x(:, 3)), '*-','DisplayName', 'partPivLU');
loglog((n), (x(:, 4)), '*-','DisplayName', 'QR');
loglog((n), (x(:, 5)), '*-','DisplayName', 'LDLt');
loglog((n), (x(:, 6)), '*-','DisplayName', 'fMatrix');
xlabel("matrix size");
ylabel("duration");
legend('show')

File: tests/pp.py
#!/usr/bin/python3 

import numpy as np 
import matplotlib.pyplot as plt


filename = "compEigenVsfMatrix"

rawData= np.loadtxt(filename, dtype='i', delimiter=' ')
#rawData = pd.readcsv(filename, header=None)
#print(rawData)
n = rawData[:,0] 

plt.figure(1)
plt.title('Execution time')
plt.plot(n, rawData[:,1], label='fullPivLU')
plt.plot(n, rawData[:,2], label='partPivLU')
plt.plot(n, rawData[:,3], label='QR')
plt.plot(n, rawData[:,4], label='LDLt')
plt.plot(n, rawData[:,5], label='fMatrix')


plt.xlabel('Matrix size')
plt.ylabel('Resolution time (s)')
plt.xscale('log')
plt.yscale('log')
plt.legend(loc='upper left')
plt.grid(visible=True)

plt.show()




x = rawData[:,1]
